pyanaconda/ui/gui/spokes/network.py

```diff
@@ -1521,11 +1521,14 @@ def apply(self):
                 log.debug("network spoke (apply) refresh payload")
 
                 from pyanaconda.payload import payloadMgr
-                if not self.payload.verifyAvailableRepositories():
+                if payloadMgr.running:
+                    log.debug("Payload is in the process of restarting, skip the repo availability check.")
```

Comment:
This problem is more general. It could happen even without this changes, because this part in `payload/__init__.py`.
```
# If a restart thread is already running, don't start a new one 
if threadMgr.get(THREAD_PAYLOAD_RESTART):
    return
```
So even without my test, it won't be able to run payload thread again and as a bonus it just silently skip restart of the payload.

Main question is, if it is a problem. I hope that DNF is able to somehow react on change of connectivity in a middle of this process. However if we want to solve this potential issue we can for example block network spoke in the middle of restarting.

So or so I think it shouldn't be part of this PR.